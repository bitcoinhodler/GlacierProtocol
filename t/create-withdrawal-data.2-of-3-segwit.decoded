{
  "txid": "80480de8573e0a7bdf2f9e43e0e936b37d0f489c9881c278ae826acc58470df2",
  "hash": "c4f0e5e4a833929169e2ae51ddc1de42e9e9fe7030a38381915ae2f366cb21c2",
  "version": 2,
  "size": 374,
  "vsize": 184,
  "weight": 734,
  "locktime": 0,
  "vin": [
    {
      "txid": "d1dd805a4f5f6ed6e32de40099f16ec8473f74819c6fa60e39bdcfd43e19c658",
      "vout": 3,
      "scriptSig": {
        "asm": "0020d017965b0cae52a8b6f6e43b00118be27dae84b825959dcb77ca6b5310e53ea3",
        "hex": "220020d017965b0cae52a8b6f6e43b00118be27dae84b825959dcb77ca6b5310e53ea3"
      },
      "txinwitness": [
        "",
        "30440220457aa54b3b30d659def503c8eafc8613f839395448fbbae568f4080e2b157b0502205b59c3eae31d7efd94388c139870858563355cc4ad8d13ae806390508431c46501",
        "3044022002c4e40edd2a1bdbc157d5ce6c88af29ae897befd56b1c2c4b066dcbefcb5eec022027c32cc0b53538f7eb1b3753ec57af4a447083b3e73f3cac2d4214faec31fce201",
        "5221035a0cf2b8ad46945154d80b339f730ac0cdbc39a95550a95821adf6df6e6e3c9421038f339e9149fda8496360d689b5d6b4d66f8e64e28b1c89846efd0831512eab882103442945263f31819baf5799dc9595eba49b8f6674dadf21189f717abd630ab15053ae"
      ],
      "sequence": 4294967295
    }
  ],
  "vout": [
    {
      "value": 0.19998160,
      "n": 0,
      "scriptPubKey": {
        "asm": "OP_DUP OP_HASH160 b6c7733e23a271e57ca16e5cd885174917968db9 OP_EQUALVERIFY OP_CHECKSIG",
        "desc": "addr(mxBQD1QAYpwiudaCJdRhE9QSW9cokafJ99)#fxf0ktmw",
        "hex": "76a914b6c7733e23a271e57ca16e5cd885174917968db988ac",
        "address": "mxBQD1QAYpwiudaCJdRhE9QSW9cokafJ99",
        "type": "pubkeyhash"
      }
    }
  ]
}
