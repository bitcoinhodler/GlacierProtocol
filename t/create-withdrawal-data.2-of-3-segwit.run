#!/bin/bash
set -e

$GLACIERSCRIPT --testnet=$1 create-withdrawal-data << INPUT
y
y
y
y
y
y
2MvQKEWTdtH7uM5C72quuMukWNFkFYjm34N
5221035a0cf2b8ad46945154d80b339f730ac0cdbc39a95550a95821adf6df6e6e3c9421038f339e9149fda8496360d689b5d6b4d66f8e64e28b1c89846efd0831512eab882103442945263f31819baf5799dc9595eba49b8f6674dadf21189f717abd630ab15053ae
mxBQD1QAYpwiudaCJdRhE9QSW9cokafJ99
1
02000000015701865854493f0cb97b07ccf231003150433c74abc8cdac4c3c87fb25bbe9e0000000006a473044022003061e39e0eafff6120261e1930da298d14d46e594de1cf260cb7ef18446d3d3022010ff3990751a8e9cb90698223ca67607706a6d670ad9d1f63b55b560c73ab65a012102d69841fccc853bc99a1a32514d53d950528bd0eae03f45107cc10ce1ed4845acfeffffff05002d31010000000017a914fdd200f6e02076173292642fd352dc45f849070e8790409700000000001976a91414f909762e0f653521433c3d853d1f90dad17ee188ac002d31010000000017a91497c2ffdcdfc233a328751b46a47b781b1eec9b2d87002d31010000000017a91422a07fe0ea8b8293eb336b9423f7e3958917924387002d31010000000017a9142524a7e29329a636bf4c1d8dea0dc6a087e5d91687bd911300
2
cMvAmArzxkXMh8k5FcaRWLBA2SgDSc2U8q1YE5hSLSek1GuyFBP3
cPSsBu9SyNVAS2Evy3m4ELFx7KGnudH3N77Es83nafa2xVWJGRSe
10
y

y
INPUT
