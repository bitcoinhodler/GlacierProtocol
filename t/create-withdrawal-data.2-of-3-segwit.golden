Are you running this on a computer WITHOUT a network connection of any kind? (y/n)?Have the wireless cards in this computer been physically removed? (y/n)?Are you running on battery power? (y/n)?Are you running on an operating system booted from a USB drive? (y/n)?Is your screen hidden from view of windows, cameras, and other people? (y/n)?Are smartphones and all other nearby devices turned off and in a Faraday bag? (y/n)?
You will need to enter several pieces of information to create a withdrawal transaction.


*** PLEASE BE SURE TO ENTER THE CORRECT DESTINATION ADDRESS ***


Source cold storage address: 
Redemption script for source cold storage address: 
Destination address: 
How many unspent transactions will you be using for this withdrawal? 
Please paste raw transaction #1 (hexadecimal format) with unspent outputs at the source address
OR
input a filename located in the current directory which contains the raw transaction data
(If the transaction data is over ~4000 characters long, you _must_ use a file.):

Transaction data found for source address.
TOTAL unspent amount for this raw transaction: 0.20000000 BTC

How many private keys will you be signing this transaction with (at least 2 required to complete transaction)?
#: Key #1: Key #2: 
Enter fee rate.
Satoshis per vbyte: 
Based on the provided rate, the fee will be 0.00002160 bitcoin.
Confirm? (y/n): 
Please enter the decimal amount (in bitcoin) to withdraw to the destination address.

Example: For 2.3 bitcoins, enter "2.3".

After a fee of 0.00002160, you have 0.19997840 bitcoins available to withdraw.

*** Technical note for experienced Bitcoin users:  If the withdrawal amount & fee are cumulatively less than the total amount of the unspent transactions, the remainder will be sent back to the same cold storage address as change. ***

Amount to send to mxBQD1QAYpwiudaCJdRhE9QSW9cokafJ99 (leave blank to withdraw all funds stored in these unspent transactions): With no change output, the transaction fee is reduced, and 0.19998160 BTC will be sent to your destination.

Is this data correct?
*** WARNING: Incorrect data may lead to loss of funds ***

0.20000000 BTC in unspent inputs from cold storage address 2MvQKEWTdtH7uM5C72quuMukWNFkFYjm34N
0.19998160 BTC going to destination address mxBQD1QAYpwiudaCJdRhE9QSW9cokafJ99
Fee amount: 0.00001840

Signing with private keys: 
cMvAmArzxkXMh8k5FcaRWLBA2SgDSc2U8q1YE5hSLSek1GuyFBP3
cPSsBu9SyNVAS2Evy3m4ELFx7KGnudH3N77Es83nafa2xVWJGRSe


Confirm? (y/n): 
Calculating transaction...

Final fee rate: 10.00 satoshis per vbyte

Raw signed transaction (hex):
0200000000010158c6193ed4cfbd390ea66f9c81743f47c86ef19900e42de3d66e5f4f5a80ddd10300000023220020d017965b0cae52a8b6f6e43b00118be27dae84b825959dcb77ca6b5310e53ea3ffffffff01d0253101000000001976a914b6c7733e23a271e57ca16e5cd885174917968db988ac04004730440220457aa54b3b30d659def503c8eafc8613f839395448fbbae568f4080e2b157b0502205b59c3eae31d7efd94388c139870858563355cc4ad8d13ae806390508431c46501473044022002c4e40edd2a1bdbc157d5ce6c88af29ae897befd56b1c2c4b066dcbefcb5eec022027c32cc0b53538f7eb1b3753ec57af4a447083b3e73f3cac2d4214faec31fce201695221035a0cf2b8ad46945154d80b339f730ac0cdbc39a95550a95821adf6df6e6e3c9421038f339e9149fda8496360d689b5d6b4d66f8e64e28b1c89846efd0831512eab882103442945263f31819baf5799dc9595eba49b8f6674dadf21189f717abd630ab15053ae00000000

Transaction fingerprint (md5):
b3d7aabd2aa47a9341dd26a3e0e1f788
QR code for transaction written to transaction.png
